disperser/dataapi/service_availability_handler.go

```diff
@@ -0,0 +1,112 @@
+package dataapi
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/health/grpc_health_v1"
+)
+
+var (
+	mutex sync.Mutex
+)
+
+func (s *server) getServiceAvailability(ctx context.Context, hosts []string) ([]*ServiceAvailability, error) {
+
+	if hosts == nil {
+		return nil, fmt.Errorf("hostnames cannot be nil")
+	}
+
+	availaiblityStatuses := make([]*ServiceAvailability, len(hosts))
+
+	for i, host := range hosts {
+		pool, ok := s.getClientPool(host)
+		if !ok {
+			return nil, fmt.Errorf("Invalid hostname: %s", host)
+		}
+		conn, err := getClientConn(pool)
+		if err != nil {
+			return nil, fmt.Errorf("Error getting client connection: %v", err)
+		}
+		defer conn.Close()
+		client := grpc_health_v1.NewHealthClient(conn)
```

Comment:
Do you plan to continue to use sidecar to serve this check?

Reply:
I think the goal need to include removing the sidecar for Disperser and Churner for a simpler solution.